### PR TITLE
Point cloud examples

### PR DESCRIPTION
**Pull request by gnavvy**
point cloud examples for both ply and laz format.

orbit-control.js is the same as in the 3D surface Plot Example.
files inside the worker folder can be ignored. (will es6-ify some of them if time permitting)

### CODE REVIEW

**Comment by shaojingli** [comment]
Is it intended that point-cloud-ply doesn't have an app.js?

File: examples/point-cloud-laz/app.js [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+      viewport: {
+        ...viewport,
+        rotationY: viewport.rotationY + 1
+      }
+    });
+    window.requestAnimationFrame(this._onUpdate);
+  }
+
+  _renderLazPointCloudLayer() {
+    return this.state.points.length && new PointCloudLayer({
```
Comment (ibgreen) [inline examples/point-cloud-laz/app.js]: Can this check be avoided?

File: examples/point-cloud-laz/app.js [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+      radiusPixels: 1
+    });
+  }
+
+  _renderDeckGLCanvas() {
+    const {width, height, viewport} = this.state;
+    const canvasProps = {width, height, ...viewport};
+    const glViewport = OrbitController.getViewport(canvasProps);
+
+    return width && height && <OrbitController {...canvasProps} ref={canvas => {
```
Comment (ibgreen) [inline examples/point-cloud-laz/app.js]: Nit: the line break formatting here isn't very pretty. I'd use a parenthesis around the component, put it on its own line, and then one prop per line.

File: examples/point-cloud-laz/app.js [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+      this._canvas = canvas;
+    }} onChangeViewport={this._onChangeViewport}>
+      <DeckGL width={width} height={height} viewport={glViewport}
```
Comment (ibgreen) [inline examples/point-cloud-laz/app.js]: One prop per line could be easier to read?

File: examples/point-cloud-laz/worker/laslaz.js [hunk 4]
```diff
@@ -0,0 +1,368 @@
+/* eslint no-use-before-define: 0 */
```
Comment (ibgreen) [inline examples/point-cloud-laz/worker/laslaz.js]: No way to install this as a module? Does it all have to be copied?

File: examples/point-cloud-ply/utils/orbit-control.js [hunk 5]
```diff
@@ -0,0 +1,177 @@
+/* global window */
```
Comment (ibgreen) [inline examples/point-cloud-ply/utils/orbit-control.js]: Didn't I see another orbit controller above?

**Comment by gnavvy** [comment]
good catch! not intended, it slipped.

File: examples/point-cloud-laz/worker/laslaz.js (hunk 4 above)
Comment (gnavvy) [inline examples/point-cloud-laz/worker/laslaz.js]: plas.io, where the decoder originally comes from, did not publish it as a module.
we can polish the code, address the source and put it on npm for sure.
but it is going to take awhile. shall we hold this PR for that?

File: examples/point-cloud-ply/utils/orbit-control.js (hunk 5 above)
Comment (gnavvy) [inline examples/point-cloud-ply/utils/orbit-control.js]: yes, two separate examples. we talked about to consolidate all the viewport + interaction logics, after that I can just import the orbit controller.

**Comment by gnavvy** [comment]
> A lot of duplicated code (that is already duplicated from other places, like OrbitControls). 
> I wonder if we should organize this as one folder that exports two demos?
I agree we should reuse the code as much as possible. But in this case, I deliberately separate them out into two examples. PLY point-cloud (mesh generated) and LAS/LAZ point-cloud (3D scan) have different use cases plus loading and parsing logics.

I see the following two parts can be shared:
1. the position normalization logics, mostly the same.
2. the orbit-control, again, will import instead of duplicate the code once it is part of core deck.gl.

File: examples/point-cloud-laz/app.js (hunk 1 above)
Comment (gnavvy) [inline examples/point-cloud-laz/app.js]: you mean to create a PointCloudLayer even if the data is empty?

**Comment by gnavvy** [comment]
just noticed resolving conflicts using the GitHub UI will result in a merge instead of rebase.

File: examples/point-cloud-laz/worker/laslaz.js (hunk 4 above)
Comment (ibgreen) [inline examples/point-cloud-laz/worker/laslaz.js]: Probably not a blocker but let's discuss.

**Pull request merged**